presto-main/src/main/java/com/facebook/presto/metadata/PolymorphicScalarFunction.java

```diff
@@ -115,19 +119,61 @@ public ScalarFunctionImplementation specialize(BoundVariables boundVariables, in
         List<Object> extraParameters = computeExtraParameters(matchingMethodsGroup.get(), context);
         MethodHandle matchingMethodHandle = applyExtraParameters(matchingMethod.get(), extraParameters);
 
-        return new ScalarFunctionImplementation(nullableResult, nullableArguments, matchingMethodHandle, deterministic);
+        return new ScalarFunctionImplementation(nullableResult, nullableArguments, getIsNullAnnotations(matchingMethod.get()),
+                matchingMethodHandle, deterministic);
+    }
+
+    private static boolean hasIsNullAnnotation(Annotation[] annotations)
```

Comment:
I spend so much time to make this work! It was main challenge. It seems like it is good that is supports unboxing now. And uniform. One of possibilities what to pass now Long to binaryOpeartorNullable to make this work.